server/src/main/java/org/jetbrains/bsp/bazel/server/sync/ProjectFileParser.kt

```diff
@@ -0,0 +1,50 @@
+package org.jetbrains.bsp.bazel.server.sync
+
+import com.google.protobuf.TextFormat
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.runBlocking
+import org.jetbrains.bsp.bazel.info.BspTargetInfo
+import java.net.URI
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.attribute.FileTime
+import kotlin.io.path.getLastModifiedTime
+
+class ProjectFileParser {
+
+    data class TimeStampInfo(val timestamp: FileTime, val bspTargetInfo: BspTargetInfo.TargetInfo)
+
+    private val timestampsMap: HashMap<URI, TimeStampInfo> = hashMapOf()
+
+    fun readTargetMapFromAspectOutputs(files: Set<URI>): Map<String, BspTargetInfo.TargetInfo> =
+        runBlocking(Dispatchers.IO) {
+            files.asFlow().map(::getTargetInfoForFileTimeStamp).toList()
+                .associateBy(BspTargetInfo.TargetInfo::getId)
+        }
+
+    private fun readTargetInfoFromFile(uri: URI): BspTargetInfo.TargetInfo {
```

Comment:
maybe import `BspTargetInfo.TargetInfo` to just write `TargetInfo` everywhere instead of `BspTargetInfo.TargetInfo`